src/pebble.py

```diff
@@ -97,6 +97,8 @@ def change_config(self, container: ops.model.Container) -> None:
                 synapse.enable_federation_domain_whitelist(
                     container=container, charm_state=self._charm_state
                 )
+            if self._charm_state.synapse_config.allow_public_rooms_over_federation:
+                synapse.enable_allow_public_rooms_over_federation(container=container)
```

Comment:
What is the reason for putting this in the pebble service please? I realise there are other similar interactions above

I thought that the pebble service is more for generating the pebble service rather than specifically reacting to every event? It doesn't seem like the interactions here are related to pebble, they seem to be related to handling events from the charm?

Reply:
Any configuration change requires a Pebble Layer replan so that's why this is here. This way the database observer and the config-change event call the same method, avoiding repetitive code.

Discussed here:
https://github.com/canonical/synapse-operator/pull/13#discussion_r1255151747

Reply:
So wouldn't it be possible to handle `config-changed` in `charm.py` and the ask pebble to restart the service as a part of the code in `charm.py`? I might have missed it, I don't see any circular dependency issues here since all of the interactions with the synapse module just depend on the container which is available in `charm.py`

Reply:
Could you create an issue/story for that so we can discuss a solution? Changing this in this PR would affect how the configuration change is handled, being something out of context here.

Reply:
Yes we can make it a separate story

Reply:
Thanks!